libs/wingc/src/jsify.rs

```diff
@@ -348,7 +348,7 @@ impl<'a> JSifier<'a> {
 							InterpolatedStringPart::Expr(e) => {
 								match *self.get_expr_type(e) {
 									Type::Json | Type::MutJson => {
-										format!("${{JSON.stringify({}, null, 2)}}", self.jsify_expression(e, ctx))
+										format!("${{typeof {} === 'string' ? {} : JSON.stringify({}, null, 2)}}", self.jsify_expression(e, ctx), self.jsify_expression(e, ctx), self.jsify_expression(e, ctx))
```

Comment:
Expressions run code. If the expression has side-effects, emitting it multiple times can be an issue. For example, if the expression is `doSomeCrazyExpensiveCalculation()` or `send1gOfDataToTheNetwork()`, we can't afford to call it multiple times.

An alternative approach would be to evaluate it only ones and then use it in the ternary.

Something like this:

```
((e) => typeof e === 'string' ? e : JSON.stringify(e, null, 2))({expr})
```